docs/authentication/jwt-auth.html

```diff
@@ -16,13 +16,407 @@
+-H 'Authorization: Bearer &lt;Your JWT String here&gt;'</code></pre>
+<p>You should see your JWT with your username returned to you. This should look something like:</p>
+<pre><code>{"claims":{"iss":"Kitura","sub":"Joe Bloggs","exp":574703307.61258602},"header":{"typ":"JWT","alg":"ES256"}}</code></pre>
+<p>Congratulations! You have just created a JWT single sign on system using a Kitura Server.
+     Your completed `JWTRoutes.swift` file for hs256 should look as follows:</p>
+<pre><code class="language-swift">import KituraContracts
+import SwiftJWT
+import Foundation
+
+
+func initializeJWTRoutes(app: App) {
+
+    app.router.post("/jwtlogin") { request, response, next in
+        let credentials = try request.read(as: UserCredentials.self)
+        let myClaims = ClaimsStandardJWT(iss: "Kitura", sub: credentials.username, exp: Date(timeIntervalSinceNow: 3600))
+        var myJWT = JWT(claims: myClaims)
+        let signedJWT = try myJWT.sign(using: App.jwtSigner)
+        response.send(signedJWT)
+        next()
+    }
+
+    app.router.get("/jwtprotected") { request, response, next in
+        let authHeader = request.headers["Authorization"]
+            guard let authComponents = authHeader?.components(separatedBy: " "),
+                authComponents.count == 2,
+                authComponents[0] == "Bearer",
+                let jwt = try? JWT&lt;ClaimsStandardJWT&gt;(jwtString: authComponents[1], verifier: App.jwtVerifier)
+                else {
+                    let _ = response.send(status: .unauthorized)
+                    return try response.end()
+            }
+        response.send(jwt)
+        next()
+    }
+}
+
+extension App {
+    // Example for HMAC signer and verifier
+    static let jwtSigner = JWTSigner.hs256(key: Data("kitura".utf8))
+    static let jwtVerifier = JWTVerifier.hs256(key: Data("kitura".utf8))
+}</code></pre>
+    <div class="underline"></div>
+    <h2 class="heading-2"><span class="blue-text">Step 9:</span> JWTs on Codable Routes (Optional)</h2>
+    <p class="block-text">In our example we used raw routing since we wanted to access the request headers.
+         If we want to use JWTs on our codable routes, we need to encapulate this code in a TypeSafeMiddleware.</p>
+    <h2 class="heading-2"><span class="blue-text">Step 9a:</span> Define our TypeSafeJWT middleware.</h2>
+    <p>Create a new file called `TypeSafeJWT.swift` in `Sources` > `Application` > `Middlewares`</p>
+    <p>Inside this file, define `TypeSafeJWT` with the following code:</p>
+<pre><code class="language-swift">import SwiftJWT
+import Kitura
+
+struct TypeSafeJWT&lt;C: Claims&gt;: TypeSafeMiddleware {
+    static func handle(request: RouterRequest, response: RouterResponse, completion: @escaping (TypeSafeJWT<C>?, RequestError?) -> Void) {
+
+    }
+}</code></pre>
+    <p class="block-text">The `handle` function is required for a TypeSafeMiddleware and is where we will interact with the headers.
```

Comment:
"The `TypeSafeMiddleware` protocol requires us to implement the `handle` function. The `handle` function is where we will interact with the request headers; we then return an instance of `TypeSafeJWT` on success, or a `RequestError` on failure. The `TypeSafeJWT` struct is generic so it can be used on any JWT."